stdlib/SparseArrays/src/linalg.jl

```diff
@@ -150,8 +125,8 @@ function mul!(C::StridedVecOrMat, X::AdjOrTransStridedMatrix, adjA::Adjoint{<:An
     end
     C
 end
-*(X::AdjOrTransStridedMatrix{TX}, adjA::Adjoint{<:Any,<:AbstractSparseMatrixCSC{TvA,TiA}}) where {TX,TvA,TiA} =
-    (T = promote_op(matprod, TX, TvA); mul!(similar(X, T, (size(X, 1), size(adjA, 2))), X, adjA, one(T), zero(T)))
+*(X::AdjOrTransStridedMatrix, adjA::Adjoint{<:Any,<:AbstractSparseMatrixCSC{TvA,TiA}}) where {TvA,TiA} =
+    (T = promote_op(matprod, eltype(X), eltype(adjA)); mul!(similar(X, T, (size(X, 1), size(adjA, 2))), X, adjA, true, false))
```

Comment:
If you're not trying to force specialization, you could even get rid of the `TvA, TiA` parameters (here and below). But not important.

Reply:
Absolutely, missed that, thanks

Reply:
If you're interested in getting rid of them, there are quite a few (just FYI). I like that you're getting rid of unnecessary parameters because it's easier to read the code, but they were there before, so entirely optional!

Reply:
Yes, I noticed, going through that right now, thanks again!

Reply:
... though I think I'll leave the type parameters in `AbstractSparseMatrixCSC` and `StridedArray`, as they should always match the eltype, correct? That way it should be clear why we call eltype in the other cases. A matter of taste, probably unimportant.

Reply:
Do you use those parameters? If you're not using them, not needing to force specialization, and not trying to enforce diagonal dispatch, e.g.,

```julia
foo(A::SparseMatrixCSC{TA,Ti}, B::SparseMatrixCSC{TB,Ti}) where {TA,TB,Ti}   # this requires Ti to be the same for A and B
```
then including them in the signature shouldn't do anything.

Reply:
What I mean is that something like
`*(adjA::Adjoint{<:Any,<:AbstractSparseMatrixCSC}, x::StridedVector{Tx}) where {Tx} =  (T = promote_op(matprod, eltype(adjA), Tx); ...)` could also be written as `*(adjA::Adjoint{<:Any,<:AbstractSparseMatrixCSC}, x::StridedVector) =  (T = promote_op(matprod, eltype(adjA), eltype(x)); ...)`, because the `eltype` always matches the type parameter for `StridedVector` (unlike e.g. adjoint vectors). Both versions  are probably identical in lowered code and performance (are they?), so it's probably a matter of taste. Which one would you prefer? The first probably makes it clearer *why* there are eltype calls all over the place in the other methods.